packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -107,29 +117,87 @@ class TlsValidationAcmTrust extends TlsValidationTrust {
   }
 }
 
-class TlsValidationFileTrust extends TlsValidationTrust {
+class TlsValidationFileTrust extends MutualTLSAuthEligibleTlsValidationTrust {
   /**
    * Path to the Certificate Chain file on the file system where the Envoy is deployed.
    */
   readonly certificateChain: string;
 
-  constructor (props: TlsValidationFileTrustOptions) {
+  constructor (certificateChain: string) {
     super();
-    this.certificateChain = props.certificateChain;
+    this.certificateChain = certificateChain;
   }
 
   public bind(_scope: Construct): TlsValidationTrustConfig {
     return {
-      virtualNodeClientTlsValidationTrust: {
+      tlsValidationTrust: {
         file: {
           certificateChain: this.certificateChain,
         },
       },
-      virtualGatewayClientTlsValidationTrust: {
-        file: {
-          certificateChain: this.certificateChain,
+    };
+  }
+}
+
+class TlsValidationSdsTrust extends MutualTLSAuthEligibleTlsValidationTrust {
+  /**
+   * The name of the secret for Envoy to fetch from a specific endpoint through the Secrets Discovery Protocol.
+   */
+  readonly secretName: string;
+
+  constructor (secretName: string) {
+    super();
+    this.secretName = secretName;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    return {
+      tlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
         },
       },
     };
   }
 }
+
+/**
+ * All Properties for Subject Alternative Names Matcher for both Client Policy and Listener.
+ */
+export interface SubjectAlternativeNamesMatcherConfig {
+  /**
+   * VirtualNode CFN configuration for subject alternative names secured by the certificate.
+   */
+  readonly subjectAlternativeNamesMatch: CfnVirtualNode.SubjectAlternativeNameMatchersProperty;
+}
+
+/**
+ * Used to generate Subject Alternative Names Matchers
+ */
+export abstract class SubjectAlternativeNames {
+  /**
+   * The values of the SAN must match the specified values exactly.
+   *
+   * @param subjectAlternativeNames The exact values to test against.
+   */
+  public static exactMatch(subjectAlternativeNames: string[]): SubjectAlternativeNames {
```

Comment:
1. Let's call this `matchingExactly()`.
2. Let's have this a variadic argument (`...names: string[])`.

Reply:
> Let's call this `matchingExactly()`.
Addressed.

> Let's have this a variadic argument (`...names: string[]`).
Thanks for bringing up this recommendation! I am fairly new to typescript and trying to learn the benefit of using the variadic argument here to accept unknown number of `string[]`s. Is this the best practice hired in CDK project or is it considered user-friendly?

Reply:
Yes, it saves you from having to create an array yourself in that instance.

Reply:
Thank you! I really love this  design.